app/forms/mappings/Mappings.scala

```diff
@@ -34,8 +38,40 @@ trait Mappings extends Formatters with Constraints {
                         invalidKey: String = "error.boolean"): FieldMapping[Boolean] =
     of(booleanFormatter(requiredKey, invalidKey))
 
-
   protected def enumerable[A](requiredKey: String = "error.required",
                               invalidKey: String = "error.invalid")(implicit ev: Enumerable[A]): FieldMapping[A] =
     of(enumerableFormatter[A](requiredKey, invalidKey))
+
+  protected def schemeTypeMapping(requiredTypeKey: String, invalidTypeKey: String,
+                                  requiredOtherKey: String, invalidOtherKey: String): Mapping[SchemeType] = {
+
+    def fromSchemeType(schemeType: SchemeType): (String, Option[String]) = {
+      schemeType match {
+        case SchemeType.Other(someValue) => ("Other", Some(someValue))
+        case _ => (schemeType.toString, None)
+      }
+    }
+
+    def toSchemeType(schemeTypeTuple: (String, Option[String])): SchemeType = {
+
+      val mappings: Map[String, SchemeType] = Seq(
+        SingleTrust,
+        GroupLifeDeath,
+        BodyCorporate
+      ).map(v => (v.toString, v)).toMap
+
+      schemeTypeTuple match {
+        case ("Other", Some(value)) => Other(value)
```

Comment:
could this be `case (key, Some(value)) if key == SchemeType.Other.toString => ...` then we won't be hardcoding the expected value.